Splinxy working on

Splinxyy opened the pull request:
I added some descriptions that I got from http://bulbapedia.bulbagarden.net/wiki/Charizard_(Pok%C3%A9mon)#Pok.C3.A9dex_entries_2 , not sure if they are actual descriptions or not.

I also fixed some bugged out things, where it had the class pasted twice and such.

MarcusTalbotNL commented on src/com/github/splinxyy/textbasedpokemon/entities/Pokemon.java:
```diff
@@ -1,215 +1,115 @@
-	protected ExperienceCategory expCategory;
-
-	protected Map<String, Move> moveMap;
-	
```
I do kinda need all those maps/lists, or at least the variables, as they are used in various calculations in the GameplauUtilities-class. I'm very aware they are a mess, but could we keep them in for now, until there's a better solution?

MarcusTalbotNL commented on src/com/github/splinxyy/textbasedpokemon/utils/GameplayUtilities.java:
```diff
@@ -1,151 +1,117 @@
-		});
-		System.out.print("\n");
-	}
-
```
The methods below this are all in line with the algorithms/formulas described on bulbapedia. Mind keeping them in for now?

MarcusTalbotNL commented on src/com/github/splinxyy/textbasedpokemon/utils/GameplayUtilities.java:
```diff
@@ -1,151 +1,117 @@
+			for(int i = 0; i < effect.size(); i++){
+				if(random < effect.get(i).getCatchThreshold()) {
+					return true;
+				}
```
Should probably add (as this is in line with the algorithm described here: http://bulbapedia.bulbagarden.net/wiki/Catch_rate): 
<code>else if(random - effect.get(i).getCatchThreshhold() > pokemon.getCatchRate()) return false;</code>

MarcusTalbotNL commented on src/com/github/splinxyy/textbasedpokemon/utils/GameplayUtilities.java:
```diff
@@ -1,151 +1,117 @@
+				if(random < effect.get(i).getCatchThreshold()) {
+					return true;
+				}
+			}
```
You close the else-statement here, so to eclipse's knowledge there is a possiblity the method does not return a boolean value. If you don't close it, the <code>return false;</code> at the end shouldn't be necessary anymore.

MarcusTalbotNL commented on src/com/github/splinxyy/textbasedpokemon/utils/GameplayUtilities.java:
```diff
@@ -1,151 +1,117 @@
+				}
+			}
+			
+			Random r = new Random();
```
Instead of using a new Random and int, we could use the GameplayUtilities.NUMBER_GENERATOR, and the previously declared random int, so that random = GameplayUtilities.NUMBER_GENERATOR.nextInt(effect.size());

MarcusTalbotNL commented on src/com/github/splinxyy/textbasedpokemon/utils/GameplayUtilities.java:
```diff
@@ -1,151 +1,117 @@
+			
+			Random r = new Random();
+			int rand = r.nextInt(effect.size());
+			for(int i = 0; i < effect.size(); i++){
```
This for-loop would become rendundant if you add the functionality in the previous for-loop using the else-if-statement mentioned earlier.

MarcusTalbotNL commented on src/com/github/splinxyy/textbasedpokemon/utils/GameplayUtilities.java:
```diff
@@ -1,151 +1,117 @@
+				if(i!=rand) continue;
+				if((random - effect.get(i).getCatchThreshold()) > pokemon.getCatchRate()) {
+					return false;
+				} else {
```
No need for this else-statement once the else-if is implemented either, since it will either return, or just continue.

Splinxyy commented:
I might have removed some while fixing something. When I updated from master, there was two classes inside one class with identical names, so I just removed the bottom one - maybe I should've removed the top. Fix it up with what you need then push.

MarcusTalbotNL commented:
Will do, thanks!
